examples/lib/cryptconverter.rb

```diff
@@ -0,0 +1,95 @@
+require 'openssl'
+
+module Temporal
+  module Client
+    module Converter
+      class Crypt < Base
+        CIPHER = 'aes-256-gcm'.freeze
+        GCM_NONCE_SIZE = 12
+        GCM_TAG_SIZE = 16
+
+        METADATA_KEY_ID_KEY = 'encryption-key-id'.freeze
+        METADATA_ENCODING_KEY = 'encoding'.freeze
+        METADATA_ENCODING = 'binary/encrypted'.freeze
+
+        def get_key_id
+          'test'
+        end
+
+        def get_key(_key_id)
+          "test-key-test-key-test-key-test!".b
+        end
+
+        def to_payloads(data)
+          key_id = get_key_id
+          key = get_key(key_id)
+
+          payloads = super(data)
+
+          Temporal::Api::Common::V1::Payloads.new(
+            payloads: payloads.payloads.map { |payload| encrypt_payload(payload, key_id, key) }
```

Comment:
So with this approach the mime type stays the same even though the payload gets replaced with a crypt, right? Is this how other SDKs do this as well? I'd expect something about the encoding to tell if this has been encrypted or not, like "application/json+encrypted" or something. Otherwise not using the right converter will result in really odd errors

Reply:
The metadata of the encrypted payload is 'binary/encrypted' https://github.com/coinbase/temporal-ruby/pull/56/files#diff-c6da22fe11b91519511828bd740132ec6b7aa0c54d50bd2a03c65242782a5b01R62. It doesn't matter what the content type is inside because the whole inner payload (metadata + data) is serialised + encrypted. After decryption we have the original payload back (with the original content type in metadata, `application/json` or whatever) which is then passed to the normal composite converter.